### PR TITLE
Minor fix in care plan generation

### PR DESCRIPTION
**Pull request by qaziabubakar-vd**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] or Closes #[issue number]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 

**Code Reviewer Checklist**
- [x] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [x] I have verified any strings visible on UI components are in the `strings.xml` file
- [x] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [x] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by maimoonak — DISMISSED** [review]
Kindly add an issue for technical debt and add along with TODO comment in code.

**Review by pld — CHANGES_REQUESTED** [review]
please a test that fails without this change and passes with it

**Review by ellykits — APPROVED** [review]

**Comment by ellykits** [comment]
@qaziabubakar Kindly address @pld  comment on test so we can have this merged.

**Comment by qaziabubakar-vd** [comment]
> @qaziabubakar Kindly address @pld comment on test so we can have this merged.

Working on it, will update the PR with test and share for review

File: android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt [hunk 1]
```diff
@@ -145,14 +145,20 @@ constructor(
...
           definition.dynamicValue.forEach { dynamicValue ->
             if (definition.kind == ActivityDefinition.ActivityDefinitionKind.CAREPLAN)
-              dynamicValue.expression.expression
+              dynamicValue
+                .expression
+                .expression
                 .let { fhirPathEngine.evaluate(null, input, planDefinition, subject, it) }
+                ?.takeIf { it.isNotEmpty() }
                 ?.let { evaluatedValue ->
+                  // Todo handle cases where we explicitly need to set previous value as null, when
```
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt:154]:
```suggestion
                  // TODO handle cases where we explicitly need to set previous value as null, when
```

**Review by pld — DISMISSED** [review]
🎉

File: android/engine/src/test/java/org/smartregister/fhircore/engine/task/FhirCarePlanGeneratorTest.kt [hunk 2]
```diff
@@ -1137,6 +1138,28 @@ class FhirCarePlanGeneratorTest : RobolectricTest() {
       }
   }
 
+  @Test
+  fun `test generateOrUpdateCarePlan returns success even when evaluatedValue is null`() =
```
Comment (pld) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/task/FhirCarePlanGeneratorTest.kt:1142]:
```suggestion
  @Test
  @ExperimentalCoroutinesAp
  fun `test generateOrUpdateCarePlan returns success even when evaluatedValue is null`() =
```

**Review by pld — APPROVED** [review]

File: android/engine/src/test/java/org/smartregister/fhircore/engine/task/FhirCarePlanGeneratorTest.kt [hunk 3]
```diff
@@ -1137,6 +1138,29 @@ class FhirCarePlanGeneratorTest : RobolectricTest() {
       }
   }
 
+  @Test
+  @ExperimentalCoroutinesAp
+  fun `test generateOrUpdateCarePlan returns success even when evaluatedValue is null`() =
```
Comment (pld) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/task/FhirCarePlanGeneratorTest.kt:1143]:
```suggestion
  @ExperimentalCoroutinesApi
  fun `test generateOrUpdateCarePlan returns success even when evaluatedValue is null`() =
```

**Review by pld — APPROVED** [review]

**Pull request merged**